fxa/_utils.py

```diff
@@ -83,23 +81,59 @@ def scope_matches(provided, required):
     if not isinstance(required, (list, tuple)):
         required = [required]
 
-    def split_subscope(s):
-        return tuple((s.split(':') + [None])[:2])
+    for req in required:
+        if not any(_match_single_scope(prov, req) for prov in provided):
+            return False
+
+    return True
+
 
-    provided = set([split_subscope(p) for p in provided])
-    required = set([split_subscope(r) for r in required])
+def _match_single_scope(provided, required):
+    if provided.startswith('https:'):
+        return _match_url_scope(provided, required)
+    else:
+        return _match_shortname_scope(provided, required)
 
-    root_provided = set([root for (root, sub) in provided])
-    root_required = set([root for (root, sub) in required])
 
-    if not root_required.issubset(root_provided):
+def _match_shortname_scope(provided, required):
+    if required.startswith('https:'):
         return False
+    provNames = provided.split(':')
+    reqNames = required.split(':')
```

Comment:
How come all the new stuff is camelCase? the rest of the file seems to be using `_` ?

Reply:
Nice catch, I'll adjust for consistency.